cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/writer/MetadataMessage.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.metadata.writer;
+
+import co.cask.cdap.common.conf.Constants;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+
+/**
+ * A container for messages send to the metadata topic configured by {@link Constants.Metadata#MESSAGING_TOPIC}.
+ * It carries the message type and the payload as {@link JsonElement}.
+ */
+public final class MetadataMessage {
+
+  /**
+   * The message type.
+   */
+  public enum Type {
+    LINEAGE,
+    USAGE
+  }
+
+  private final Type type;
+  private final JsonElement payload;
```

Comment:
why not use `DatasetUsage` directly?

Reply:
This class is a container for different types of metadata, including lineage, usage, workflow tokens and user metadata. Each of them has different structure